File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] t) {
        Stack<Integer> stk = new Stack<>();
        for(int i = 0 ; i < t.length ; i++){
            if(t[i].equals("+") || t[i].equals("-") || t[i].equals("*") || t[i].equals("/")){
                int v2 = stk.pop();
                int v1 = stk.pop();
                if(t[i].equals("+")){
                    stk.push(v1+v2);
                }else if(t[i].equals("-")){
                    stk.push(v1-v2);
                }else if (t[i].equals("*")){
                    stk.push(v1*v2);
                }else if (t[i].equals("/")){
                    stk.push(v1/v2);
                }
            }else{
                int val = Integer.parseInt(t[i]);
                stk.push(val);
            }
        }
        return stk.pop();
    }
}

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.kt
class Solution {
    fun evalRPN(tokens: Array<String>): Int {
        val stk : MutableList<Int> = mutableListOf()
        for(ele in tokens){
            if(ele.equals("*") || ele.equals("/") || ele.equals("+") || ele.equals("-")){
                val n1 = stk.removeAt(stk.size-1);
                val n2 = stk.removeAt(stk.size-1);
                when(ele){
                    "*" -> stk.add(n2*n1)
                    "/" -> stk.add(n2/n1)
                    "+" -> stk.add(n2+n1)
                    "-" -> stk.add(n2-n1)
                }
            }else{
                stk.add(Integer.parseInt(ele))
            }
        }
        return stk[0]
    }
}